core/fields/Field_Honeypot.php

```diff
@@ -0,0 +1,66 @@
+<?php
+defined('CMSPATH') or die; // prevent unauthorized access
+
+class Field_Honeypot extends Field {
+
+	function __construct($default_content="") {
+		$this->id = "";
+		$this->name = "";
+		$this->default = $default_content;
+		$this->content_type="";
+		$this->nowrap = true;
+		$this->save=false;
+	}
+
+	public function display() {
+		// autocomplete attribute set to nonsense which is calculated to be same as 'off' without being explicit
+		// tabindex is required as -1 for accessibility reasons, so might tip off some bots, but can't hurt screen readers etc
+		// set value to be ' ' (space) - allows us to use 'required' client-side, but whitespace might be more enticing to replace 
+		// set display/style attributes via js as another layer of obfuscation
+		?>
+		<input required placeholder="Important information" type='text' tabindex="-1" autocomplete="nothingtoseehere" id='<?php echo $this->id;?>' <?php echo $this->get_rendered_name();?> value=' '/>
+		<script>
+			let hp = document.getElementById('<?php echo $this->id;?>') ?? null;
+			if (hp) {
+				hp.style.position = 'absolute'; // take up no space in document flow
+				hp.style.clipPath = 'circle(0)'; // sneaky invisible method that might hide from most bots
```

Comment:
should put the comments in `<?php //comment ?>` so they dont show up on the front end